core/src/main/java/hudson/cli/declarative/CLIRegisterer.java

```diff
@@ -202,6 +200,7 @@ public int main(List<String> args, Locale locale, InputStream stdin, PrintStream
 
                             CmdLineParser parser = bindMethod(binders);
                             try {
+                                // TODO this could probably use ACL.as; why is it calling SecurityContext.setAuthentication rather than SecurityContextHolder.setContext?
```

Comment:
IIRC this class is no longer maintained / outdated as it's preferable for plugins to implement commands directly.

Reply:
It is not currently deprecated, but yes I would recommend `CLICommand` in all cases.